service/controller/cloudconfig/base_extension.go

```diff
@@ -28,6 +30,16 @@ func (e *baseExtension) templateData(certFiles []certs.File) templateData {
 		certsPaths = append(certsPaths, file.AbsolutePath)
 	}
 
+	primaryScaleSetName := key.WorkerVMSSName(e.customObject)
```

Comment:
This piece of code is used for both masters and workers. We only use `MachinePool` when creating workers, so we need to make sure is not `nil`.